src/interfaces/node.cpp

```diff
@@ -207,8 +207,9 @@ class NodeImpl : public Node
     bool getNetworkActive() override { return m_context->connman && m_context->connman->GetNetworkActive(); }
     CFeeRate estimateSmartFee(int num_blocks, bool conservative, int* returned_target = nullptr) override
     {
+        if (!m_context->fee_estimator) return {};
         FeeCalculation fee_calc;
-        CFeeRate result = ::feeEstimator.estimateSmartFee(num_blocks, &fee_calc, conservative);
+        CFeeRate result = m_context->fee_estimator->estimateSmartFee(num_blocks, &fee_calc, conservative);
```

Comment:
cc21167415aa01db862035cd07b6b9b35fb53f35

this is unused. Instead of shuffling code around, you may add a commit that removes it.

```diff
diff --git a/src/interfaces/node.cpp b/src/interfaces/node.cpp
index 2c5f8627e6..a9ca4f9cf8 100644
--- a/src/interfaces/node.cpp
+++ b/src/interfaces/node.cpp
@@ -205,15 +205,6 @@ public:
         }
     }
     bool getNetworkActive() override { return m_context->connman && m_context->connman->GetNetworkActive(); }
-    CFeeRate estimateSmartFee(int num_blocks, bool conservative, int* returned_target = nullptr) override
-    {
-        FeeCalculation fee_calc;
-        CFeeRate result = ::feeEstimator.estimateSmartFee(num_blocks, &fee_calc, conservative);
-        if (returned_target) {
-            *returned_target = fee_calc.returnedTarget;
-        }
-        return result;
-    }
     CFeeRate getDustRelayFee() override { return ::dustRelayFee; }
     UniValue executeRpc(const std::string& command, const UniValue& params, const std::string& uri) override
     {
diff --git a/src/interfaces/node.h b/src/interfaces/node.h
index 5079be038e..36f76aeb4f 100644
--- a/src/interfaces/node.h
+++ b/src/interfaces/node.h
@@ -151,9 +151,6 @@ public:
     //! Get network active.
     virtual bool getNetworkActive() = 0;
 
-    //! Estimate smart fee.
-    virtual CFeeRate estimateSmartFee(int num_blocks, bool conservative, int* returned_target = nullptr) = 0;
-
     //! Get dust relay fee.
     virtual CFeeRate getDustRelayFee() = 0;

Reply:
Right, i didn't notice.. Cherry-picked your patch and adapted it to the new interface path post-rebase.
```patch
Author: Antoine Poinsot <darosior@protonmail.com>
Date:   Tue Dec 1 11:24:33 2020 +0100

    interface: remove unused estimateSmartFee method from node
    
    Co-Authored-by: MarcoFalke <falke.marco@gmail.com>
    Signed-off-by: Antoine Poinsot <darosior@protonmail.com>

diff --git a/src/interfaces/node.h b/src/interfaces/node.h
index 5079be038..36f76aeb4 100644
--- a/src/interfaces/node.h
+++ b/src/interfaces/node.h
@@ -151,9 +151,6 @@ public:
     //! Get network active.
     virtual bool getNetworkActive() = 0;
 
-    //! Estimate smart fee.
-    virtual CFeeRate estimateSmartFee(int num_blocks, bool conservative, int* returned_target = nullptr) = 0;
-
     //! Get dust relay fee.
     virtual CFeeRate getDustRelayFee() = 0;
 
diff --git a/src/node/interfaces.cpp b/src/node/interfaces.cpp
index 77a5957a5..3cd9c6fda 100644
--- a/src/node/interfaces.cpp
+++ b/src/node/interfaces.cpp
@@ -221,15 +221,6 @@ public:
         }
     }
     bool getNetworkActive() override { return m_context->connman && m_context->connman->GetNetworkActive(); }
-    CFeeRate estimateSmartFee(int num_blocks, bool conservative, int* returned_target = nullptr) override
-    {
-        FeeCalculation fee_calc;
-        CFeeRate result = ::feeEstimator.estimateSmartFee(num_blocks, &fee_calc, conservative);
-        if (returned_target) {
-            *returned_target = fee_calc.returnedTarget;
-        }
-        return result;
-    }
     CFeeRate getDustRelayFee() override { return ::dustRelayFee; }
     UniValue executeRpc(const std::string& command, const UniValue& params, const std::string& uri) override
     {
```